geopy/geocoders/banfrance.py

```diff
@@ -0,0 +1,194 @@
+import warnings
+
+from geopy.compat import urlencode
+from geopy.geocoders.base import DEFAULT_SENTINEL, Geocoder
+from geopy.location import Location
+from geopy.util import logger
+
+__all__ = ("BANFrance", )
+
+
+class BANFrance(Geocoder):
+    """Geocoder using the Base Adresse Nationale France API.
+
+    Documentation at:
+        https://adresse.data.gouv.fr/api
+
+    """
+
+    geocode_path = '/search'
+    reverse_path = '/reverse'
+
+    def __init__(
+            self,
+            domain='api-adresse.data.gouv.fr',
+            scheme=None,
+            timeout=DEFAULT_SENTINEL,
+            proxies=DEFAULT_SENTINEL,
+            user_agent=None,
+            format_string=None,
+            ssl_context=DEFAULT_SENTINEL,
+    ):
+        """
+
+        :param str domain: Currently it is ``'api-adresse.data.gouv.fr'``, can
+            be changed for testing purposes.
+
+        :param str scheme:
+            See :attr:`geopy.geocoders.options.default_scheme`.
+
+        :param int timeout:
+            See :attr:`geopy.geocoders.options.default_timeout`.
+
+        :param dict proxies:
+            See :attr:`geopy.geocoders.options.default_proxies`.
+
+        :param str user_agent:
+            See :attr:`geopy.geocoders.options.default_user_agent`.
+
+            .. versionadded:: 1.12.0
+
+        :param str format_string:
+            See :attr:`geopy.geocoders.options.default_format_string`.
+
+            .. versionadded:: 1.14.0
+
+        :type ssl_context: :class:`ssl.SSLContext`
+        :param ssl_context:
+            See :attr:`geopy.geocoders.options.default_ssl_context`.
+
+            .. versionadded:: 1.14.0
+
+        """
+        super(BANFrance, self).__init__(
+            format_string=format_string,
+            scheme=scheme,
+            timeout=timeout,
+            proxies=proxies,
+            user_agent=user_agent,
+            ssl_context=ssl_context,
+        )
+        self.domain = domain.strip('/')
+
+        self.geocode_api = (
+            '%s://%s%s' % (self.scheme, self.domain, self.geocode_path)
+        )
+        self.reverse_api = (
+            '%s://%s%s' % (self.scheme, self.domain, self.reverse_path)
+        )
+
+    def geocode(
+            self,
+            query,
+            limit=None,
```

Comment:
This parameter seems to be ignored: I believe the implementation is accidentally missing.

Also the corresponding testcase is missing, could you please add one? You could use this one for inspiration: https://github.com/geopy/geopy/blob/3b19c8e4f6fdf13dddfbe7a3703ba031ec95194d/test/geocoders/nominatim.py#L48-L60